ranger/core/progress_tracker.py

```diff
@@ -0,0 +1,33 @@
+# This file is part of ranger, the console file manager.
+# License: GNU GPL version 3, see the file "AUTHORS" for details.
+
+"""
+A Progress Tracker of any Loadable that manages all the progress
+related statistics.
+"""
+
+import time
+
+class ProgressTracker(object):
+    def __init__(self, total_items=0):
+        self.total_items = total_items
+        self.done = 0
+        self.start_time = time.time()
+
+    def set_total_items(self, total_items):
+        self.total_items = total_items
+
+    def step(self, stepsize=1):
+        self.done += stepsize
+
+    @property
+    def percent(self):
+        if self.total_items == 0:
+            return 0
+        return (self.done / self.total_items) * 100
```

Comment:
If total and done are both integers, Python 2 will perform integer division here. To prevent that, multiply with `1.0` before the division.

Reply:
No, please use `from __future__ import division` instead.

Reply:
Oh, yes, please do that. I haven't used future imports in years.